staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go

```diff
@@ -82,6 +83,11 @@ type Scheme struct {
 	// schemeName is the name of this scheme.  If you don't specify a name, the stack of the NewScheme caller will be used.
 	// This is useful for error reporting to indicate the origin of the scheme.
 	schemeName string
+
+	// gvToAPILifecycle maps GroupVersion to its APILifecycle.
+	gvToAPILifecycle map[schema.GroupVersion]schema.APILifecycle
+	// gvrToAPILifecycle maps GroupVersionResource to its APILifecycle.
+	gvrToAPILifecycle map[schema.GroupVersionResource]schema.APILifecycle
```

Comment:
Possibly to be clarified in the comments here- If all GVRs have the same lifecycle as the GV, do we require a map entry for these GVRs, or do we only require a GVR map entry when the GVR has a different lifecycle than the GV?

Reply:
done